lib/nostrum/api/ratelimiter.ex

```diff
@@ -33,7 +34,10 @@ defmodule Nostrum.Api.Ratelimiter do
 
   def init([]) do
     :ets.new(:ratelimit_buckets, [:set, :public, :named_table])
-    {:ok, []}
+    domain = to_charlist(Constants.domain())
+    {:ok, conn_pid} = :gun.open(domain, 443)
```

Comment:
Remember the rules of `:gun` safety. The option for `retry` must be set to `1_000_000_000` (`:infinity` doesn’t work) for `:gun` to stay up. It defaults to `5`.

Reply:
You are absolutely right. 72ef871 makes gun invulnerable.